core/imds/src/main/java/software/amazon/awssdk/imds/Ec2Metadata.java

```diff
@@ -29,4 +34,137 @@ public interface Ec2Metadata {
      * @return Instance metadata value
      */
     String get(String path);
+
+    /**
+     * @return The retry policy which includes the number of retry attempts for any failed request.
+     */
+    RetryPolicy retryPolicy();
+
+    /**
+     * @return The endpoint of IMDS.
+     */
+    URI endpoint();
+
+    /**
+     * @return The port for the endpoint.
+     */
+    Integer port();
+
+    /**
+     * @return The Time to live (TTL) of the token.
+     */
+    Duration tokenTtl();
+
+    /**
+     * @return The endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+     */
+    String endpointMode();
+
+    /**
+     * @return The output stream for debugging.
+     */
+    String httpDebugOutput();
+
+    /**
+     * @return The number of seconds to sleep in-between retries.
+     */
+    Integer backoff();
+
+    /**
+     * @return The SdkHttpClient instance to make the http requests.
+     */
+    SdkHttpClient httpClient();
+
+    /**
+     * @return The Builder Object consisting all the fields.
+     */
+    Ec2Metadata.Builder toBuilder();
+
+    /**
+     * Create an {@code Ec2Metadata} using the default values.
+     */
+    static Ec2Metadata create() {
+        return builder().build();
+    }
+
+    /**
+     * Creates a default builder for {@link Ec2Metadata}.
+     */
+    static Ec2Metadata.Builder builder() {
+        return DefaultEc2Metadata.builder();
+    }
+
+    /**
+     * The builder definition for a {@link Ec2Metadata}.
+     */
+    interface Builder  {
+
+        /**
+         * Define the retry policy which includes the number of retry attempts for any failed request.
+         *
+         * @param retryPolicy The retry policy which includes the number of retry attempts for any failed request.
+         * @return Returns a reference to this builder
+         */
+        Builder retryPolicy(RetryPolicy retryPolicy);
+
+        /**
+         * Define the endpoint of IMDS.
+         *
+         * @param endpoint The endpoint of IMDS.
+         * @return Returns a reference to this builder
+         */
+        Builder endpoint(URI endpoint);
+
+        /**
+         * Define the port for the endpoint.
+         *
+         * @param port The port for the endpoint.
+         * @return Returns a reference to this builder
+         */
+        Builder port(Integer port);
```

Comment:
This is probably not needed because it should be supplied as part of URI in `endpoint`